engine/src/main/scala/cromwell/engine/workflow/ShadowWorkflowActor.scala

```diff
@@ -1,19 +1,124 @@
+    */
+  sealed trait StartMode
+  case object NoStartMode extends StartMode
+  case object StartNewWorkflow extends StartMode
+  case object RestartExistingWorkflow extends StartMode
+
+  /**
+    * States for the ShadowWorkflowActor FSM
+    */
+  sealed trait ShadowWorkflowActorState
+
+  /**
+    * Waiting for a Start or Restart command.
+    */
+  case object Unstarted extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor is created. It now needs to validate and create its WorkflowDescriptor
+    */
+  case object MaterializingWorkflowDescriptor extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor has a descriptor. It now needs to create its set of WorkflowBackendActors
+    */
+  case object MaterializingWorkflowBackendActors extends ShadowWorkflowActorState
+
+  /**
+    * The WorkflowActor has a descriptor and a set of backends. It now needs to execute the hell out of its jobs.
+    */
+  case object ExecutingWorkflow extends ShadowWorkflowActorState
+
+  /**
+    * We're done. But we're more than done. We're Clive Dunn.
+    */
+  case object Done extends ShadowWorkflowActorState
+
+  case class ShadowWorkflowActorData(startMode: StartMode,
+                                     backendMap: Map[Call, ActorRef])
+
+  def props(workflowId: WorkflowId, wdlSource: WdlSource): Props = Props(new ShadowWorkflowActor(workflowId, wdlSource))
 }
 
 /**
-  * Class that will [eventually] orchestrate a single workflow.
+  * Class that orchestrates a single workflow... Shadow style.
   */
-class ShadowWorkflowActor extends Actor with ActorLogging{
+class ShadowWorkflowActor(workflowId: WorkflowId, wdlSource: WdlSource) extends LoggingFSM[ShadowWorkflowActorState, ShadowWorkflowActorData] with ActorLogging{
 
   private val tag = this.getClass.getSimpleName
 
-  override def receive: Receive = {
-    case unhandledMessage: Any => log.info(s"$tag received an unhandled message: $unhandledMessage")
+  startWith(Unstarted, ShadowWorkflowActorData(NoStartMode, Map.empty))
+
+  when(Unstarted) {
+    case Event(StartCommand(sm: StartMode), data) =>
+      // TODO: Begin MaterializingWorkflowDescriptor (by outsourcing to MaterialiseWorkflowDescriptorActor)
+      goto(MaterializingWorkflowDescriptor) using data.copy(startMode = sm)
+  }
+
+  when(MaterializingWorkflowDescriptor) {
+    case Event(MaterializationSuccess(workflowDescriptor: WorkflowDescriptor), stateData) =>
+      // TODO: Begin the WorkflowBackendActor materializification!
+      val newBackendAssignments: Map[Call, ActorRef] = ???
+      val initializerActor = context.actorOf(InitializeWorkflowBackendsActor.props(tag, newBackendAssignments.values.toSeq))
+      initializerActor ! StartInitializations
+      goto(MaterializingWorkflowBackendActors) using stateData.copy(backendMap = newBackendAssignments)
+    case Event(MaterializationFailure(reason: Throwable), _) =>
+      context.parent ! ShadowWorkflowFailed(reason)
```

Comment:
Oh wait, this is in materialisation, not execution. I guess we could kill children and crash?